mslib/msui/autoplot_gui.py

```diff
@@ -0,0 +1,302 @@
+                if index != -1:
+                    config_settings["automated_plotting_vsecs"][index] = [url, layer, styles, level]
+            else:
+                if index != -1:
+                    config_settings["automated_plotting_lsecs"][index] = [url, layer, styles, level]
+
+    def refresh_sig(self, config_settings):
+        autoplot_flights = config_settings["automated_plotting_flights"]
+        if self.view == "Top View":
+            autoplot_secs = config_settings["automated_plotting_hsecs"]
+        elif self.view == "Side View":
+            autoplot_secs = config_settings["automated_plotting_vsecs"]
+        else:
+            autoplot_secs = config_settings["automated_plotting_lsecs"]
+
+        self.autoplotTreeWidget.clear()
+        for row in autoplot_flights:
+            item = QTreeWidgetItem(row)
+            self.autoplotTreeWidget.addTopLevelItem(item)
+
+        self.autoplotSecsTreeWidget.clear()
+        for row in autoplot_secs:
+            item = QTreeWidgetItem(row)
+            self.autoplotSecsTreeWidget.addTopLevelItem(item)
+
+    def remove_selected_row(self, parent, treewidget, config_settings):
+        selected_item = treewidget.currentItem()
+        if selected_item:
+            index = treewidget.indexOfTopLevelItem(selected_item)
+            if index != -1:
+                treewidget.takeTopLevelItem(index)
+                if treewidget.objectName() == "autoplotTreeWidget":
+                    config_settings["automated_plotting_flights"].pop(index)
+                if treewidget.objectName() == "autoplotSecsTreeWidget":
+                    if self.view == "Top View":
+                        config_settings["automated_plotting_hsecs"].pop(index)
+                    elif self.view == "Side View":
+                        config_settings["automated_plotting_vsecs"].pop(index)
+                    else:
+                        config_settings["automated_plotting_lsecs"].pop(index)
+            else:
+                parent = selected_item.parent()
+                if parent:
+                    parent.takeChild(parent.indexOfChild(selected_item))
+        parent.refresh_signal_emit.emit()
+
+    def combo_box_input(self, combo):
+        comboBoxName = combo.objectName()
+        currentText = combo.currentText()
+        if comboBoxName == "timeIntervalComboBox":
+            self.intv = currentText
+
+    def updateDateTimeValue(self):
+        self.stime = self.stimeSpinBox.dateTime().toString('yyyy/MM/dd HH:mm UTC')
+        self.etime = self.etimeSpinBox.dateTime().toString('yyyy/MM/dd HH:mm UTC')
+
+    def update_config_file(self, config_settings):
+        options = QFileDialog.Options()
+        options |= QFileDialog.DontUseNativeDialog
+
+        # Get the user's home directory
+        home_directory = str(Path.home())
+
+        # Set the default path
+        default_path = os.path.join(home_directory, ".config", "msui", "combined_data.json")
```

Comment:
please use
constants.MSUI_CONFIG_PATH

this is important or a testrun will chang user data